lib/src/model_utils.dart

```diff
@@ -47,26 +47,42 @@ bool isInExportedLibraries(
 
 bool isPrivate(Element e) => e.name.startsWith('_');
 
+final _privacyExpando = new Expando<bool>();
+
 bool isPublic(Element e) {
-  if (isPrivate(e)) return false;
-  // check to see if element is part of the public api, that is it does not
-  // have a '#nodoc' in the documentation comment
-  if (e is PropertyAccessorElement && e.isSynthetic) {
-    var accessor = (e as PropertyAccessorElement);
-    if (accessor.correspondingSetter != null &&
-        !accessor.correspondingSetter.isSynthetic) {
-      e = accessor.correspondingSetter;
-    } else if (accessor.correspondingGetter != null &&
-        !accessor.correspondingGetter.isSynthetic) {
-      e = accessor.correspondingGetter;
+  var cachedVal = _privacyExpando[e];
```

Comment:
This lazy-init pattern can written

bool isPublic(e) => _privacyExpando[e] ??= _isPublic(e);

where `_isPublic` is the old definition of `isPublic`.